src/duct/migrator/ragtime.clj

```diff
@@ -58,9 +58,14 @@
 (defn- add-hash-to-id [migration]
   (update migration :id str "#" (subs (hash-migration migration) 0 8)))
 
-(defn- migrate [index {:keys [database migrations strategy logger]
+(defn- migrate [index {:keys [database migrations strategy logger migrations-table]
                        :or   {strategy :raise-error}}]
-  (ragtime/migrate-all (jdbc/sql-database (:spec database)) index migrations
+  (ragtime/migrate-all (jdbc/sql-database (:spec database)
+                                          (if migrations-table
```

Comment:
That's looking a little hard to understand. Perhaps we write something more like:

```clojure
(defn- get-database [{{:keys [spec]} :database :as opts}]
  (jdbc/sql-database spec (select-keys opts [:migrations-table])))

(defn- get-strategy [{:keys [strategy] :or {strategy :raise-error}]
  (strategies strategy))

(defn- get-reporter [{:keys [logger]}]
  (if logger (logger-reporter logger) reporter/print))

(defn- migrate [index {:keys [migrations] :as opts}]
  (let [db    (get-database opts)
        strat (get-strategy opts)
        rep   (get-reporter opts)]
    (ragtime/migrate-all db index migrations {:reporter rep, :strategy strat})
    (ragtime/into-index index migrations)))
```